pointcloud/pointcloud_utils.go

```diff
@@ -91,3 +91,29 @@ func StatisticalOutlierFilter(meanK int, stdDevThresh float64) (func(PointCloud)
 	}
 	return filterFunc, nil
 }
+
+// ToBasicOctree takes a pointcloud object and converts it into a basic octree.
+func ToBasicOctree(cloud PointCloud) (*BasicOctree, error) {
+	if basicOctree, ok := cloud.(*BasicOctree); ok {
+		return basicOctree, nil
+	}
+
+	center := getCenterFromPcMetaData(cloud.MetaData())
+	maxSideLength := getMaxSideLengthFromPcMetaData(cloud.MetaData())
+	basicOctree, err := NewBasicOctree(center, maxSideLength)
+	if err != nil {
+		return &BasicOctree{}, err
+	}
+	var iterateError error
```

Comment:
cloud.Iterate returns an error. This is unneeded.

Reply:
Correction; it returns a bool. This is indeed probably how we want to surface this error. Never mind.